convert2rhel/unit_tests/subscription_test.py

```diff
@@ -488,37 +500,49 @@ def test_download_rhsm_pkgs(version, downloaded_pkgs, monkeypatch):
 
 
 @pytest.mark.parametrize(
-    ("submgr_installed", "unregistration_failure"),
-    (
-        (True, True),
-        (True, False),
-        (False, True),
-        (False, False),
-    ),
+    ("output", "ret_code", "expected"),
+    (("", 0, "System unregistered successfully."), ("Failed to unregister.", 1, "System unregistration failed")),
 )
-def test_unregister_system(submgr_installed, unregistration_failure, monkeypatch, caplog):
-    if unregistration_failure:
-        monkeypatch.setattr(utils, "run_subprocess", mock.Mock(return_value=("output", 1)))
-    else:
-        monkeypatch.setattr(utils, "run_subprocess", mock.Mock(return_value=("output", 0)))
+def test_unregister_system(output, ret_code, expected, monkeypatch, caplog):
+    submgr_command = ("subscription-manager", "unregister")
+    rpm_command = ("rpm", "--quiet", "-q", "subscription-manager")
+
+    # Mock rpm command
+    run_subprocess_mock = mock.Mock(
+        side_effect=run_subprocess_side_effect(
+            (
+                submgr_command,
+                (
+                    output,
+                    ret_code,
+                ),
+            ),
+            (rpm_command, ("", 0)),
+        ),
+    )
+    monkeypatch.setattr(utils, "run_subprocess", value=run_subprocess_mock)
 
-    if submgr_installed:
-        monkeypatch.setattr(
-            pkghandler, "get_installed_pkg_objects", lambda _: [namedtuple("Pkg", ["name"])("subscription-manager")]
-        )
-    else:
-        monkeypatch.setattr(pkghandler, "get_installed_pkg_objects", lambda _: None)
+    subscription.unregister_system()
+
+    assert expected in caplog.records[-1].message
 
+
+def test_unregister_system_submgr_not_found(monkeypatch, caplog):
+    rpm_command = "rpm --quiet -q subscription-manager"
+    run_subprocess_mock = mock.Mock(
+        side_effect=unit_tests.run_subprocess_side_effect(
+            ((rpm_command,), ("", 1)),
+        )
+    )
+    monkeypatch.setattr(utils, "run_subprocess", value=run_subprocess_mock)
     subscription.unregister_system()
+    assert "The subscription-manager package is not installed." in caplog.records[-1].message
 
-    if submgr_installed:
-        utils.run_subprocess.assert_called_once()
-        if unregistration_failure:
-            assert "System unregistration failed" in caplog.text
-        else:
-            assert "System unregistered successfully." in caplog.text
-    else:
-        assert "The subscription-manager package is not installed." in caplog.text
+
+def test_unregister_system_keep_rhsm(monkeypatch, caplog):
+    monkeypatch.setattr(tool_opts, "keep_rhsm", value=True)
+    subscription.unregister_system()
+    assert "Skipping due to the use of --keep-rhsm." in caplog.records[-1].message
 
```

Comment:
When I have a lot of pytest test functions in a single file, I like to use classes to organize them a bit.

So something like
``` python
class TestUnregisteringSystem:
    def test_unregister_system[...]
    def test_unregister_system_submgr_not_found[...]
    def test_unregister_system_keep_rhsm[...]
    def test_unregister_system_skipped[...]
```

might be nice to do.  This is an optional request, though.  It doesn't change the functioning in any way, it's purely stylistic.

Reply:
Think this is a nice way of grouping them. Adding the changes to the next commit. Thanks!